server/leader.go

```diff
@@ -191,6 +185,18 @@ func (s *Server) campaignLeader() error {
 		return errors.Trace(err)
 	}
 
+	s.enableLeader(true)
+	defer func() {
+		s.enableLeader(false)
+		// Reset connections and cluster.
+		s.closeAllConnections()
+		s.cluster.stop()
+		// Clear ts.
+		s.ts.Store(&atomicObject{
```

Comment:
seem you can set nil directly? so `s.ts.Load().(*atomicObject)` will return not ok.